common/Util.cpp

```diff
@@ -21,9 +21,11 @@
 #  include <sys/syscall.h>
 #  include <sys/vfs.h>
 #  include <sys/resource.h>
+#  include <sys/random.h>
```

Comment:
according to [https://man7.org/linux/man-pages/man2/getrandom.2.html](https://man7.org/linux/man-pages/man2/getrandom.2.html) I added `#include <sys/random.h>` but some tests fail as: `compiler couldn't find the file »sys/random.h« referenced from common/Util.cpp:24:26`

Reply:
This would mean that your system doesn't have this header. What distro are you using, and what version is it? (Notice that this function is only available on Linux.)

Reply:
The header file works fine on my system(ManjaroLinux 22.0.4). Some tests here -on Jenkins- fails and don't recognize `#include <sys/random.h>` . for example: (GCC 6.3.0 check — FAILURE)  [https://cpci.cbg.collabora.co.uk:8080/job/github_online_master-gcc-6.3.0/5611/](https://cpci.cbg.collabora.co.uk:8080/job/github_online_master-gcc-6.3.0/5611/). I just wonder why GCC couldn't find the header.

Thanks!

Reply:
Right, I meant to add this in the issue report but forgot. We need to support non-linux systems. So, we can't remove the old approach completely. We need to wrap them in `#ifdef` statement (you'll find examples in the code). Also, I think we need to check for `sys/random.h` in configure.ac so we fallback.